diff -Nru sysvinit-2.85.orig/src/Makefile sysvinit-2.85/src/Makefile
--- sysvinit-2.85.orig/src/Makefile	Tue Nov  6 12:58:16 2001
+++ sysvinit-2.85/src/Makefile	Thu Apr 24 12:05:06 2003
@@ -8,8 +8,9 @@
 # Version:	@(#)Makefile  2.83-3  06-Nov-2001  miquels@cistron.nl
 #
 
-CC	= cc
-CFLAGS	= -Wall -O2 -D_GNU_SOURCE
+CC	 = gcc
+OPTIMIZE = -O2
+CFLAGS	= -Wall $(OPTIMIZE) -D_GNU_SOURCE
 LDFLAGS	= -s
 STATIC	=
 
@@ -89,7 +90,8 @@
 			runlevel shutdown $(ROOT)/sbin
 		# These are not installed by default
 ifeq ($(DEBIAN),)
-		$(INSTALL) -m 555 utmpdump wall $(ROOT)/usr/bin
+		$(INSTALL) -m 755 wall $(ROOT)/usr/bin
+		$(INSTALL) -m 755 utmpdump  $(ROOT)/usr/bin/utmpx-dump
 endif
 		# $(INSTALL) -m 755 etc/initscript.sample $(ROOT)/etc
 		$(INSTALL) -m 755 mesg last $(ROOT)/usr/bin
diff -Nru sysvinit-2.85.orig/src/paths.h sysvinit-2.85/src/paths.h
--- sysvinit-2.85.orig/src/paths.h	Mon Apr 14 13:37:01 2003
+++ sysvinit-2.85/src/paths.h	Thu Apr 24 12:06:06 2003
@@ -24,7 +24,7 @@
 #define FORCEFSCK	"/forcefsck"		/* Force fsck on boot */
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
 #define SHELL		"/bin/sh"		/* Default shell */
-#define INITSCRIPT	"/etc/initscript"	/* Initscript. */
+#define INITSCRIPT	"/etc/rc.d/rc.init"	/* Initscript. */
 #define PWRSTAT		"/etc/powerstatus"	/* COMPAT: SIGPWR reason (OK/BAD) */
 
 #if 0
diff -Nru sysvinit-2.85.orig/src/utmpdump.c sysvinit-2.85/src/utmpdump.c
--- sysvinit-2.85.orig/src/utmpdump.c	Tue Sep 12 23:14:24 2000
+++ sysvinit-2.85/src/utmpdump.c	Thu Apr 24 12:05:06 2003
@@ -265,14 +265,14 @@
 	}
 
 	if (optind < argc) {
-		fprintf(stderr, "Utmp %sdump of %s\n", reverse ? "un" : "", argv[optind]);
+		fprintf(stderr, "Utmpx %sdump of %s\n", reverse ? "un" : "", argv[optind]);
 		if ((fp = fopen(argv[optind], "r")) == NULL) {
 			perror("Unable to open file");
 			exit(1);
 		}
 	}
 	else {
-		fprintf(stderr, "Utmp %sdump of stdin\n", reverse ? "un" : "");
+		fprintf(stderr, "Utmpx %sdump of stdin\n", reverse ? "un" : "");
 		fp = stdin;
 	}
 
