--- sysvinit-3.10/src/halt.c.orig	2024-08-01 08:41:41.156638116 +0200
+++ sysvinit-3.10/src/halt.c	2024-08-01 08:48:11.933304753 +0200
@@ -81,10 +81,11 @@ extern void write_wtmp(char *user, char
  */
 void usage(void)
 {
-	fprintf(stderr, "usage: %s [-d] [-f] [-h] [-i] [-n] [-w]%s%s\n",
+	fprintf(stderr, "usage: %s [-c] [-d] [-f] [-h] [-i] [-n] [-w]%s%s\n",
 		progname,
 		strcmp(progname, "halt") ? "" : " [-p]",
 		strcmp(progname, "reboot") ? "" : " [-k] [-m <message>]");
+        fprintf(stderr, "\t-c: ask for confirmation.\n");
 	fprintf(stderr, "\t-d: don't write a wtmp record.\n");
 	fprintf(stderr, "\t-f: force halt/reboot, don't call shutdown.\n");
 	fprintf(stderr, "\t-h: put harddisks in standby mode.\n");
@@ -214,7 +215,9 @@ int main(int argc, char **argv)
 	int do_hddown = 0;
 	int do_poweroff = 0;
 	int do_kexec = 0;
+        int do_confirm = 0;
         char *user_message = NULL;
+        char answer = 'n';
 	int c;
 	char *tm = NULL;
 
@@ -234,7 +237,7 @@ int main(int argc, char **argv)
 	/*
 	 *	Get flags
 	 */
-	while((c = getopt(argc, argv, ":ihdfm:npwkt:")) != EOF) {
+	while((c = getopt(argc, argv, ":ihdfm:npwkct:")) != EOF) {
 		switch(c) {
 			case 'n':
 				do_sync = 0;
@@ -268,6 +271,9 @@ int main(int argc, char **argv)
 			case 't':
 				tm = optarg;
 				break;
+                        case 'c':
+                                do_confirm = 1;
+                                break;
 			default:
 				usage();
 		}
@@ -300,6 +306,13 @@ int main(int argc, char **argv)
 			exit(1);
 		}
 
+		if (do_confirm) {
+			fprintf(stderr, "Are you sure you want to reboot? N/y.\n");
+			scanf("%c", &answer);
+			if (tolower(answer) != 'y')
+				exit(1);
+		}
+
 		/*
 		 *	See if we are in runlevel 0 or 6.
 		 */
