--- sysvinit-3.10/src/halt.c~	2024-07-29 15:39:12.000000000 +0200
+++ sysvinit-3.10/src/halt.c	2024-08-01 08:41:12.739971452 +0200
@@ -170,6 +170,15 @@ void do_shutdown(char *fl, int should_po
 {
 	char *args[9];
 	int i = 0;
+        char *file;
+
+        if (should_poweroff) {
+            file = strdup("/poweroff");
+        } else {
+            file = strdup("/halt");
+        }
+        close(open(file, O_CREAT|O_RDWR, 0644));
+        free(file);
 
 	args[i++] = "shutdown";
 	args[i++] = fl;
