--- sysvinit-2.88dsf/src/killall5.c.wiget	2010-03-19 15:10:55.000000000 +0100
+++ sysvinit-2.88dsf/src/killall5.c	2011-12-03 23:12:52.373139250 +0100
@@ -66,8 +66,9 @@ char *Version = "@(#)killall5 2.86 31-Ju
 
 /* Info about a process. */
 typedef struct proc {
+	char *pathname;		/* full path to executable     	  */
 	char *argv0;		/* Name as found out from argv[0] */
-	char *argv0base;	/* `basename argv[1]`		  */
+	char *argv0base;	/* `basename argv[0]`		  */
 	char *argv1;		/* Name as found out from argv[1] */
 	char *argv1base;	/* `basename argv[1]`		  */
 	char *statname;		/* the statname without braces    */
@@ -477,6 +478,7 @@ int readproc(int do_stat)
 		if (p->argv0) free(p->argv0);
 		if (p->argv1) free(p->argv1);
 		if (p->statname) free(p->statname);
+		if (p->pathname) free(p->pathname);
 		free(p);
 	}
 	plist = NULL;
@@ -616,6 +618,8 @@ int readproc(int do_stat)
 				break;
 			p->dev = st.st_dev;
 			p->ino = st.st_ino;
+			p->pathname = (char *)xmalloc(PATH_MAX);
+			readlink(path, p->pathname, PATH_MAX);
 		default:
 			break;
 		}
@@ -806,6 +810,9 @@ PIDQ_HEAD *pidof(char *prog)
 		     p->argv0[0] == '-')) {
 			ok |= (strcmp(p->statname, s) == 0);
 		}
+		        
+		if (prog[0] == '/' && p->pathname && strcmp(prog, p->pathname))
+			ok = 0;
 
 		if (ok) add_pid_to_q(q, p);
 	}
