--- sysvinit-2.86/src/Makefile.orig	2004-06-09 14:47:45.000000000 +0200
+++ sysvinit-2.86/src/Makefile	2004-12-26 22:40:47.622737944 +0100
@@ -9,7 +9,8 @@
 #
 
 CC	= gcc
-CFLAGS	= -Wall -O2 -fomit-frame-pointer -D_GNU_SOURCE
+OPTIMIZE = -O2 -fomit-frame-pointer
+CFLAGS	= -Wall $(OPTIMIZE) -D_GNU_SOURCE
 LDFLAGS	= -s
 STATIC	=
 
@@ -26,7 +27,7 @@
 ifeq ($(DISTRO),)
 BIN	+= mountpoint
 SBIN	+= sulogin bootlogd
-USRBIN	+= utmpdump wall
+USRBIN	+= utmpx-dump wall
 MAN1	+= mountpoint.1 wall.1
 MAN8	+= sulogin.8 bootlogd.8
 endif
@@ -71,7 +72,7 @@
 mountpoint:	mountpoint.o
 		$(CC) $(LDFLAGS) -o $@ mountpoint.o
 
-utmpdump:	utmpdump.o
+utmpx-dump:	utmpdump.o
 		$(CC) $(LDFLAGS) -o $@ utmpdump.o
 
 runlevel:	runlevel.o
--- sysvinit-2.86/src/paths.h.orig	2004-06-09 14:47:45.000000000 +0200
+++ sysvinit-2.86/src/paths.h	2004-12-26 22:41:21.122645184 +0100
@@ -25,7 +25,7 @@
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
 #define SHELL		"/bin/sh"		/* Default shell */
 #define SULOGIN		"/sbin/sulogin"		/* Sulogin */
-#define INITSCRIPT	"/etc/initscript"	/* Initscript. */
+#define INITSCRIPT	"/etc/rc.d/rc.init"	/* Initscript. */
 #define PWRSTAT		"/etc/powerstatus"	/* COMPAT: SIGPWR reason (OK/BAD) */
 
 #if 0
diff -Nru sysvinit-2.85.orig/src/utmpdump.c sysvinit-2.85/src/utmpdump.c
--- sysvinit-2.85.orig/src/utmpdump.c	Tue Sep 12 23:14:24 2000
+++ sysvinit-2.85/src/utmpdump.c	Thu Apr 24 12:05:06 2003
@@ -265,14 +265,14 @@
 	}
 
 	if (optind < argc) {
-		fprintf(stderr, "Utmp %sdump of %s\n", reverse ? "un" : "", argv[optind]);
+		fprintf(stderr, "Utmpx %sdump of %s\n", reverse ? "un" : "", argv[optind]);
 		if ((fp = fopen(argv[optind], "r")) == NULL) {
 			perror("Unable to open file");
 			exit(1);
 		}
 	}
 	else {
-		fprintf(stderr, "Utmp %sdump of stdin\n", reverse ? "un" : "");
+		fprintf(stderr, "Utmpx %sdump of stdin\n", reverse ? "un" : "");
 		fp = stdin;
 	}
 
