diff -urN sysvinit-2.78.org/src/Makefile sysvinit-2.78/src/Makefile
--- sysvinit-2.78.org/src/Makefile	Thu Mar  2 18:17:48 2000
+++ sysvinit-2.78/src/Makefile	Thu Mar  2 18:18:08 2000
@@ -8,8 +8,9 @@
 # Version:	@(#)Makefile  2.78  13-Nov-1999  miquels@cistron.nl
 #
 
-CC	= cc
-CFLAGS	= -Wall -O2 -D_GNU_SOURCE
+CC	 = gcc
+OPTIMIZE = -O2
+CFLAGS	= -Wall $(OPTIMIZE) -D_GNU_SOURCE
 LDFLAGS	= -s
 STATIC	=
 
@@ -88,7 +89,8 @@
 			runlevel shutdown $(ROOT)/sbin
 		# These are not installed by default
 ifeq ($(DEBIAN),)
-		$(INSTALL) -m 555 utmpdump wall $(ROOT)/usr/bin
+		$(INSTALL) -m 755 wall $(ROOT)/usr/bin
+		$(INSTALL) -m 755 utmpdump  $(ROOT)/usr/bin/utmpx-dump
 endif
 		# $(INSTALL) -m 755 etc/initscript.sample $(ROOT)/etc
 		$(INSTALL) -m 755 mesg last $(ROOT)/usr/bin
diff -urN sysvinit-2.78.org/src/paths.h sysvinit-2.78/src/paths.h
--- sysvinit-2.78.org/src/paths.h	Thu Mar  2 18:17:48 2000
+++ sysvinit-2.78/src/paths.h	Thu Mar  2 18:18:08 2000
@@ -28,7 +28,7 @@
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
 #define IOSAVE		"/etc/ioctl.save"	/* termios settings for SU */
 #define SHELL		"/bin/sh"		/* Default shell */
-#define INITSCRIPT	"/etc/initscript"	/* Initscript. */
+#define INITSCRIPT	"/etc/rc.d/rc.init"	/* Initscript. */
 #if 0
 #define HALTSCRIPT1	"/etc/init.d/halt"	/* Called by "fast" shutdown */
 #define HALTSCRIPT2	"/etc/rc.d/rc.0"	/* Called by "fast" shutdown */
diff -urN sysvinit-2.78.org/src/utmpdump.c sysvinit-2.78/src/utmpdump.c
--- sysvinit-2.78.org/src/utmpdump.c	Thu Mar  2 18:17:48 2000
+++ sysvinit-2.78/src/utmpdump.c	Thu Mar  2 18:18:08 2000
@@ -263,14 +263,14 @@
 	}
 
 	if (optind < argc) {
-		fprintf(stderr, "Utmp %sdump of %s\n", reverse ? "un" : "", argv[optind]);
+		fprintf(stderr, "Utmpx %sdump of %s\n", reverse ? "un" : "", argv[optind]);
 		if ((fp = fopen(argv[optind], "r")) == NULL) {
 			perror("Unable to open file");
 			exit(1);
 		}
 	}
 	else {
-		fprintf(stderr, "Utmp %sdump of stdin\n", reverse ? "un" : "");
+		fprintf(stderr, "Utmpx %sdump of stdin\n", reverse ? "un" : "");
 		fp = stdin;
 	}
 
