--- sysvinit-2.98/src/init.c.orig	2020-11-27 22:45:28.430317602 +0100
+++ sysvinit-2.98/src/init.c	2020-11-28 08:02:50.399146908 +0100
@@ -1490,7 +1490,7 @@
 			 *	See if we have a single user entry.
 			 */
 			for(old = newFamily; old; old = old->next)
-				if (strpbrk(old->rlevel, "S"))  break;
+				if (strpbrk(old->rlevel, "S") && !ISPOWER(old->action)) break;
 			if (old == NULL)
 				snprintf(buf, sizeof(buf), "~~:S:wait:%s\n", SULOGIN);
 			else
