--- sysvinit-2.88dsf/doc/Propaganda.wiget	2009-09-10 10:28:49.000000000 +0200
+++ sysvinit-2.88dsf/doc/Propaganda	2011-12-03 22:25:33.574721688 +0100
@@ -46,7 +46,7 @@ Some other general changes:
   /var/run/utmp (it's also shown on the command line if you do a "ps").
 - unreckognized options are silently ignored (such as the infamous
   "ro" - mount root file system read only).
-- if the file /etc/initscript is present it will be used to launch
+- if the file /etc/rc.d/rc.init is present it will be used to launch
   all programs that init starts (so that you can set a generic
   umask, ulimit eg for ALL processes - see initscript.sample).
 - A "sulogin" program added that always asks for the root
--- sysvinit-2.88dsf/src/initscript.sample.wiget	2009-09-10 10:28:49.000000000 +0200
+++ sysvinit-2.88dsf/src/initscript.sample	2011-12-03 22:25:33.574721688 +0100
@@ -1,14 +1,14 @@
 #
-# initscript	If this script is intalled as /etc/initscript,
+# initscript	If this script is intalled as /etc/rc.d/rc.init,
 #		it is executed by init(8) for every program it
 #		wants to spawn like this:
 #
-#		/bin/sh /etc/initscript <id> <level> <action> <process>
+#		/bin/sh /etc/rc.d/rc.init <id> <level> <action> <process>
 #
 #		It can be used to set the default umask and ulimit
 #		of all processes. By default this script is installed
 #		as /etc/initscript.sample, so to enable it you must
-#		rename this script first to /etc/initscript.
+#		rename this script first to /etc/rc.d/rc.init.
 #
 # Version:	@(#)initscript 1.10 10-Dec-1995 MvS.
 #
--- sysvinit-2.88dsf/src/paths.h.wiget	2010-04-11 11:30:27.000000000 +0200
+++ sysvinit-2.88dsf/src/paths.h	2011-12-03 22:30:54.881898668 +0100
@@ -34,7 +34,7 @@
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
 #define SHELL		"/bin/sh"		/* Default shell */
 #define SULOGIN		"/sbin/sulogin"		/* Sulogin */
-#define INITSCRIPT	"/etc/initscript"	/* Initscript. */
+#define INITSCRIPT	"/etc/rc.d/rc.init"	/* Initscript. */
 #define PWRSTAT_OLD	"/etc/powerstatus"	/* COMPAT: SIGPWR reason (OK/BAD) */
 #define PWRSTAT		"/var/run/powerstatus"	/* COMPAT: SIGPWR reason (OK/BAD) */
 
