--- sysvinit-2.78/src/init.c.foo	Wed Apr  4 01:42:27 2001
+++ sysvinit-2.78/src/init.c	Wed Apr  4 01:42:49 2001
@@ -2451,6 +2451,8 @@
   	p = argv[0];
   	
 
+  umask(022);
+	
   /*
    *	Is this telinit or init ?
    */
@@ -2523,8 +2525,6 @@
   /* Check syntax. */
   if (argc - optind != 1 || strlen(argv[optind]) != 1) Usage(p);
   if (!strchr("0123456789SsQqAaBbCcUu", argv[optind][0])) Usage(p);
-
-  umask(022);
 
   /* Open the fifo and write a command. */
   memset(&request, 0, sizeof(request));
