diff -ru sysvinit-2.86/src/init.c sysvinit-2.86/src/init.c
--- sysvinit-2.86/src/init.c	2008-01-23 17:10:45.000000000 -0500
+++ sysvinit-2.86/src/init.c	2008-01-23 16:26:07.000000000 -0500
@@ -2671,7 +2671,7 @@
 		maxproclen += strlen(argv[f]) + 1;
 	}
 
-  	if (getenv("SELINUX_INIT") == NULL) {
+  	if (getenv("SELINUX_INIT") == NULL && !is_selinux_enabled()) {
 	  putenv("SELINUX_INIT=YES");
 	  if (selinux_init_load_policy(&enforce) == 0 ) {
 	    execv(myname, argv);
